src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -393,9 +556,117 @@ private static DataGroup convertStarTableToDataGroup(StarTable table,
         }
         dataGroup.trimToSize();
         return dataGroup;
+    }*/
+
+
+    //This function is based on the packageValue function from the FitsStarTable class in the uk.ac.starlink.fits package
+    private static Object getValAsObject(Object base, int icol, boolean[] hasBlank, long[] blanks, boolean[] isScaled,
+                                         double[] scales, double[] zeros) {
+        if (base == null) {
+            return null;
+        } else {
+            Class bcls = base.getClass();
+            Class cls = bcls.isArray() ? bcls.getComponentType() : null;
+            if (cls != null && Array.getLength(base) == 1) {
+                boolean hasblank = hasBlank[icol];
+                long blank = blanks[icol];
+                boolean scaled = isScaled[icol];
+                double scale = scales[icol];
+                double zero = zeros[icol];
+                if (cls == Byte.TYPE) {
+                    byte val = ((byte[])(base))[0];
+                    /*
+                    TODO: Attempt to use ternary operator here instead
+                     */
+                    if (hasblank && val == (byte)((int)blank))
+                        return null;
+                    if (scaled)
+                        return Double.valueOf((double)val * scale + zero);
+                    else
+                        return Byte.valueOf(val);
+                }
```

Comment:
You might want to try a switch. Combine with the other suggestions. This would make the whole thing look less verbose and be easier to read.
```java
switch (cls.toString()) {
                    case "byte" -> {
                        byte val = ((byte[])(elem))[0];
                        if (hasblank && val == blank) return null;
                        return scaled ? val * scale + zero : val;
                    }
                    case "short" -> {
                        short val = ((short[])(elem))[0];
                        if (hasblank && val == blank) return null;
                        return scaled ? val * scale + zero : val;
                    }
                }
```

Reply:
no in this case, string is `"string java.lang.String"`

Reply:
More comments. I think blanks only applies to `byte`, `int`, `long` you don't need it for `double` and `float`

Reply:
ok, better idea.

write a record class:
```java
    record EvalVal(long blank, boolean scaled, boolean hasblank, double scale, double zero) {
        public Number evalValue(Number val) {
            if (hasblank && val.doubleValue() == blank) return null;
            if (!scaled) return val;
            return val.doubleValue() * scale + zero;
        }
    }
```

in `getValAsObject` (line 582)
```java
EvalVal evaluator= new EvalVal(blanks[icol], isScaled[icol], hasBlank[icol], scales[icol], zeros[icol]);
```

every switch (float example):
```java
 return evaluator.evalValue(((float[])(elem))[0]);
```


```

Reply:
Thanks, great suggestion. Code looks a lot simpler now for `getValAsObject`.